subs/pantry/src/Pantry.hs

```diff
@@ -626,7 +628,35 @@ loadFromURL
   => Text -- ^ url
   -> Maybe BlobKey
   -> RIO env ByteString
-loadFromURL = undefined
+loadFromURL url Nothing = do
+  mcached <- withStorage $ loadURLBlob url
+  case mcached of
+    Just bs -> return bs
+    Nothing -> loadWithCheck url $ \_ -> return ()
+loadFromURL url (Just bkey@(BlobKey sha size)) = do
+  mcached <- withStorage $ loadBlob bkey
+  case mcached of
+    Just bs -> return bs
+    Nothing -> loadWithCheck url $ \bs -> do
+      let blobSha = mkStaticSHA256FromBytes bs
+          blobSize = FileSize $ fromIntegral $ B.length bs
+      when (blobSha /= sha || blobSize /= size) $
+        throwIO $ BadKey . utf8BuilderToText $
+          "Expected SHA/size : " <> display sha <> "/" <> display size <>
+          ". Actual SHA/size: " <> display blobSha <> "/" <> display blobSize
+
+loadWithCheck
+  :: (HasPantryConfig env, HasLogFunc env)
+  => Text -- ^ url
+  -> (ByteString -> RIO env ()) -- ^ function to check downloaded blob
+  -> RIO env ByteString
+loadWithCheck url checkResponseBody = do
+  req <- parseRequest $ T.unpack url
+  res <- httpLbs req
```

Comment:
We should probably check that this is a `200 OK` status code.

Reply:
and now it fails because of 404 as  https://raw.githubusercontent.com/commercialhaskell/stack-templates/master/lts/10/1.yaml doesn't appear to be a valid snapshot URL

Reply:
Typo on my part, it should be using `stackage-templates`.